libraries/targets/arnold.mtlx

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+  All rights reserved.  See LICENSE.txt for license.
+-->
+
+<materialx version="1.38">
+
+  <!-- ======================================================================== -->
+  <!-- Derived target definition for Arnold                                     -->
+  <!-- ======================================================================== -->
+  <targetdef name="arnold" inherit="genosl"/>
```

Comment:
These are the new targetdefs elements that define a new target. A targetdef has to be loaded before any shader generation starts, in order to map implementations to nodedefs for a target application/renderer.

I put these into separate files so a client can easily select which targets to support by choosing which targetdef files to load. By default in our applications all targetdefs inside libraries/targets are loaded.

Reply:
Just curious if we should add glsl, osl, mdl, genosl, targets into stdlib.mtlx so it always get's loaded without any change to existing integrations? 
Then only the non-core ones would be added to the targets directory?

Reply:
It's a bit unfortunate that an existing integration now has to make a change to make sure the appropriate target support gets loaded. However I think it might be best if that happens, to force a client to select which targets to support. 

Because then we can make sure to only load in implementations that are really needed. For example, for a GLSL client it's a waste of performance to have all implementations for glsl, osl and mdl loaded when only glsl is used. Not only in the first pass when it's loaded in, but having all these loaded in the map for each successive query of implementations.

I think a "breaking" change like this is OK since it's a move to a new version 1.38, and we don't have a wide spread use of MaterialX anyway yet. So better to make this change now than later.

Keeping it like this also make target loading consistent, and avoid classifying some targets as "core" and others as not as important. It's better if the API client can make that decision.

Reply:
The fact that I load in all targetdefs that are available in the "targets" folder is just a simplification for now. We can later make this more specific for our various back-ends and only load in the targets that are needed.